ocaml/sdk-gen/java/autogen/xen-api/src/main/java/com/xensource/xenapi/Connection.java

```diff
@@ -29,228 +29,211 @@
+ * Represents a connection to a XenServer. Creating a new instance of this class initialises a new JsonRpcClient that is
+ * then used by all method calls: each method call in xen-api takes a Connection as a parameter, composes a JSON-RPC
  * method call, and dispatches it on the Connection's client via the dispatch method.
  */
-public class Connection
-{
+public class Connection {
+    private final JsonRpcClient client;
     private APIVersion apiVersion;
-
-    /**
-     * Default reply timeout for xml-rpc calls in seconds
-     */
-    protected static final int DEFAULT_REPLY_TIMEOUT = 600;
-
     /**
-     * Default connection timeout for xml-rpc calls in seconds
+     * The opaque reference to the session used by this connection
      */
-    protected static final int DEFAULT_CONNECTION_TIMEOUT = 5;
+    private String sessionReference;
 
     /**
-     * Updated when Session.login_with_password() is called.
+     * Creates a connection to a particular server using a custom implementation of the JsonRpcClient.
+     * <p>
+     * Note this constructor does NOT call Session.loginWithPassword; the programmer is responsible for calling it,
+     * passing the Connection as a parameter. No attempt to connect to the server is made until login is called.
+     * <p>
+     * When this constructor is used, a call to dispose() will do nothing. The programmer is responsible for manually
+     * logging out the Session.
+     *
+     * @param jsonRpcClient The JsonRpcClient used to connect to the JSON-RPC backed.
      */
-    public APIVersion getAPIVersion()
-    {
-        return apiVersion;
+    public Connection(JsonRpcClient jsonRpcClient) {
+        this.client = jsonRpcClient;
     }
 
     /**
-     * The opaque reference to the session used by this connection
-     */
-    private String sessionReference;
-
-    /**
-     * As seen by the xmlrpc library. From our point of view it's a server.
+     * Creates a connection to a particular server using a given url. This object can then be passed
+     * in to any other API calls.
+     * <p>
+     * Note this constructor does NOT call Session.loginWithPassword; the programmer is responsible for calling it,
+     * passing the Connection as a parameter. No attempt to connect to the server is made until login is called.
+     * <p>
+     * When this constructor is used, a call to dispose() will do nothing. The programmer is responsible for manually
+     * logging out the Session.
+     *
+     * @param httpClient     The HttpClient used to make calls, this will be used by JsonRpcClient for handling requests
+     * @param url            The URL of the server to connect to. Should be of the form http(s)://host-url./jsonrpc or http(s)://host-url.
+     * @param requestTimeout The reply timeout for JSON-RPC calls in seconds
      */
-    private final XmlRpcClient client;
+    public Connection(CloseableHttpClient httpClient, URL url, int requestTimeout) {
```

Comment:
We may want to follow the example of the C# SDK and deprecate the overloads that pass the requestTimeout or connectionTimeout, and instead provide public setters. The reason is that if the user wants to override other parameters like the keepAlive etc., adding public setters is probably easier than adding constructor overloads with a train of parameters.